src/main/scala/devices/debug/Debug.scala

```diff
@@ -722,6 +721,263 @@ class TLDebugModuleInner(device: Device, getNComponents: () => Int, beatBytes: I
+                               (SBCSWrData.sbaccess === 1.U) && (SBCSFieldsReg.sbaccess16  =/= 1.U) ||
+                               (SBCSWrData.sbaccess === 2.U) && (SBCSFieldsReg.sbaccess32  =/= 1.U) ||
+                               (SBCSWrData.sbaccess === 3.U) && (SBCSFieldsReg.sbaccess64  =/= 1.U) ||
+                               (SBCSWrData.sbaccess === 4.U) && (SBCSFieldsReg.sbaccess128 =/= 1.U)  
+
+        when (~io.dmactive) {
+          SBCSFieldsReg := SBCSFieldsRegReset
+        }.otherwise {
+          SBCSFieldsReg.sbbusyerror     := Mux(SBCSWrEn && SBCSWrData.sbbusyerror,                   false.B, // W1C
+                                           Mux(anyAddressWrEn && SBCSFieldsReg.sbbusy,               true.B, // Set if a write to SBADDRESS occurs while busy
+                                           Mux((anyDataRdEn || anyDataWrEn) && SBCSFieldsReg.sbbusy, true.B, SBCSFieldsReg.sbbusyerror))) // Set if any access to SBDATA occurs while busy
+          SBCSFieldsReg.sbreadonaddr    := Mux(SBCSWrEn, SBCSWrData.sbreadonaddr   , SBCSFieldsReg.sbreadonaddr)
+          SBCSFieldsReg.sbautoincrement := Mux(SBCSWrEn, SBCSWrData.sbautoincrement, SBCSFieldsReg.sbautoincrement)
+          SBCSFieldsReg.sbreadondata    := Mux(SBCSWrEn, SBCSWrData.sbreadondata   , SBCSFieldsReg.sbreadondata)
+          SBCSFieldsReg.sberror         := Mux(SBCSWrEn && SBCSWrData.sberror =/= 0.U, 0.U, // W1C
+                                           Mux(SBCSWrEn && SBAlignmentError,           3.U, SBCSFieldsReg.sberror))
+        }
+       
+        cover(SBCSFieldsReg.sbbusyerror,                              "SBCS Cover", "sberror set")
+        cover(SBCSFieldsReg.sbbusy === 3.U,                           "SBCS Cover", "sbbusyerror alignment error")
+        cover(SBCSFieldsReg.sbaccess === 0.U && !SBAlignmentError,    "SBCS Cover", "8-bit access")
+        cover(SBCSFieldsReg.sbaccess === 1.U && !SBAlignmentError,    "SBCS Cover", "16-bit access")
+        cover(SBCSFieldsReg.sbaccess === 2.U && !SBAlignmentError,    "SBCS Cover", "32-bit access")
+        cover(SBCSFieldsReg.sbaccess === 3.U && !SBAlignmentError,    "SBCS Cover", "64-bit access")
+        cover(SBCSFieldsReg.sbaccess === 4.U && !SBAlignmentError,    "SBCS Cover", "128-bit access")
+        cover(SBCSFieldsReg.sbautoincrement && SBCSFieldsReg.sbbusy,  "SBCS Cover", "Access with autoincrement set")
+        cover(!SBCSFieldsReg.sbautoincrement && SBCSFieldsReg.sbbusy, "SBCS Cover", "Access without autoincrement set")
+
+        SBCSRdData             := SBCSFieldsReg
+        SBCSRdData.sbversion   := 1.U(1.W) // This code implements a version of the spec after January 1, 2018
+        SBCSRdData.sbbusy      := (sb2tl.module.io.sbStateOut =/= SystemBusAccessState.Idle.id.U)
+        SBCSRdData.sbasize     := sb2tl.module.edge.bundle.addressBits.U
+        SBCSRdData.sbaccess128 := (sb2tl.module.edge.bundle.dataBits == 128).B
```

Comment:
can you use the already computed "has128" and such?

Reply:
I could've done "sbaccess128 := hasSBData2and3", but I still have to access dataBits for the others since those aren't needed above. I figured for code clarity this would be nicer

Reply:
yeah makes sense